core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -58,19 +78,31 @@ public StandardEncryptionManager(
    * @param kmsClient Client of KMS used to wrap/unwrap keys in envelope encryption
    * @param writerKekTimeout timeout of kek (key encryption key) cache entries
    */
-  public StandardEncryptionManager(
-      String tableKeyId, int dataKeyLength, KeyManagementClient kmsClient, long writerKekTimeout) {
+  StandardEncryptionManager(
```

Comment:
I removed this from `public` since it will probably change in the next PR.

Rather than having an externally controlled "cache" I think it makes sense for the encryption manager to handle key management. So I've added an argument to pass `WrappedEncryptionKey` in (from `TableMetadata` in the next PR, probably). This now indexes the list and maintains current. We will need to merge the result of `keys` with the list in `TableMetadata` in the next steps.